### PR TITLE
feat(breadcrumb): add breadcrumb component

### PR DESCRIPTION
**Pull request by angela-1**
**Notice**
I added the devdepencency `vue-router` to support the `to` props and route.

The Composition API does not has `this`, so it does not has `this.$router`. It use `provide/inject` to get router and route to path.

[https://github.com/vuejs/rfcs/blob/master/active-rfcs/0013-composition-api.md#plugin-development](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0013-composition-api.md#plugin-development)

> Many Vue plugins today inject properties onto this. For example, Vue Router injects this.$route and this.$router, and Vuex injects this.$store. This has made type inference tricky since each plugin requires the user to augment the Vue typing for injected properties.
When using the Composition API, there is no this. Instead, plugins will leverage provide and inject internally and expose a composition function. The following is hypothetical code for a plugin:

**Problem**
When I test the code, it show a warning, maybe caused by not import the vue-router when test? I'm not sure.
```
 console.warn node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:39
      [Vue warn]: injection "Symbol([vue-router]: router)" not found.
        at <ElBreadcrumbItem>
        at <ElBreadcrumb separator-class="el-icon-check" >
        at <Anonymous ref="VTU_COMPONENT" >
        at <VTUROOT>
```

### CODE REVIEW

**Comment by a631807682** [comment]
Thank you for your PR, I will review it in this week.

File: packages/breadcrumb/src/breadcrumb-item.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ></i>
+    <span v-else class="el-breadcrumb__separator" role="presentation">{{
+      separator
+    }}</span>
+  </span>
+</template>
+<script lang="ts">
+import { defineComponent, onMounted, inject, ref, Ref } from 'vue'
+import { propsSymbol } from './propsKey'
+import { useRouter } from 'vue-router'
```
Comment (a631807682) [inline packages/breadcrumb/src/breadcrumb-item.vue:23]: If we import `vue-router` in component, we can not add it just into the devdepencency.
We should change the API here.
When user use this component, they should set `globalProperties` to pass the `router` plugin to compoent.
```
   // user component
   VueApp.config.globalProperties.$router = router
   ...

    // our  component
    const instance = getCurrentInstance();
    const router = instance.proxy.$router

   // our logic
    ...
```
Here is an example [runtime-core/src/componentProxy.ts](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/componentProxy.ts#L44)

Comment (angela-1) [reply packages/breadcrumb/src/breadcrumb-item.vue]: I understand.
the user should add `vue-router-next` dep and add the config below by himself.
So the breadcrumb component can work.

```js
const router = createRouter(/*...*/)

VueApp.use(router)
VueApp.config.globalProperties.$router = router
```

File: packages/breadcrumb/src/breadcrumb-item.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  },
+  setup(props, ctx) {
+    const separator = ref('')
+    const separatorClass = ref('')
+    const link: Ref<HTMLElement | null> = ref(null)
+
+    const propsData: any = inject(propsSymbol)
+
+    const instance = getCurrentInstance()
+    const router = (instance.proxy as any).$router
```
Comment (a631807682) [inline packages/breadcrumb/src/breadcrumb-item.vue:45]: We should warning if user haven't config.

File: test/unit/specs/breadcrumb.spec.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <el-breadcrumb-item>活动详情</el-breadcrumb-item>
+      </el-breadcrumb>
+      `,
+    })
+    await wait()
+    expect(
+      wrapper.find<HTMLElement>('.el-breadcrumb__separator.el-icon-check')
+    ).toBeTruthy()
+  })
+})
```
Comment (a631807682) [inline test/unit/specs/breadcrumb.spec.ts:37]: Is there an easy way to test with `vue-router` ? Or mock `globalProperties` to test the link click ?

Comment (angela-1) [reply test/unit/specs/breadcrumb.spec.ts:37]: Sorry!

After search on web, I can not find a method to mock `globalProperties `.
the `vue-test-utils-next` has a merged pr [https://github.com/vuejs/vue-test-utils-next/pull/83](https://github.com/vuejs/vue-test-utils-next/pull/83) And It's API does has method to mock `globalProperties `

Comment (a631807682) [reply test/unit/specs/breadcrumb.spec.ts:37]: I will try to do it later, thanks.

**Pull request merged**